sos/plugins/sssd.py

```diff
@@ -28,12 +29,17 @@ def setup(self):
             "/etc/sssd/conf.d/*.conf"
         ])
 
-        self.add_cmd_output("sssctl config-check")
-
-        domain = self.collect_cmd_output("sssctl domain-list")
-        if domain['status'] == 0:
-            for domain_name in domain['output'].splitlines():
-                self.add_cmd_output("sssctl domain-status -o " + domain_name)
+        # call sssctl commands only when sssd service is running,
+        # otherwise the command timeouts
+        sssd_pred = SoSPredicate(self, services=["sssd"])
+        if self.test_predicate(self, pred=sssd_pred):
+            self.add_cmd_output("sssctl config-check")
+
+            domain = self.collect_cmd_output("sssctl domain-list")
```

Comment:
Same here - `collect_cmd_output()` accepts a predicate so we could key off `domain['status'] == None` here to detect predicate failure and gate the domain stuff on that as well as `['status'] == 0`. Otherwise an early return on !predicate might look neater.

Reply:
Havent realized this trick (what previously caused this implementation to be denied). PR updated accordingly.